scipy/stats/_distn_infrastructure.py

```diff
@@ -3836,3 +3884,423 @@ def get_distribution_names(namespace_pairs, rv_base_class):
+    it follows a negative binomial distribution with parameters *n* and *p*\.
+    (See `scipy.stats.nbinom`.) We believe that the parameter *n* was fewer
+    than 30, and we know that the parameter *p* must lie on the interval
+    [0, 1]. We record this information in a variable `shape_bounds` and pass
+    this information to `fit`.
+
+    >>> shape_bounds = [(0, 30), (0, 1)]
+    >>> res = stats.fit(dist, data, shape_bounds)
+
+    `fit` searches within the user-specified `shape_bounds` for the
+    values that best match the data (in the sense of maximum likelihood
+    estimation). In this case, it found shape values similar to those
+    from which the data were actually generated.
+
+    >>> res.params
+    FitShapes(n=5.0, p=0.5028157644634368, loc=0.0)  # may vary
+
+    We can visualize the results by superposing the probability mass function
+    of the distribution (with the shapes fit to the data) over a normalized
+    histogram of the data.
+
+    >>> res.plot()
+
+    Note that the estimate for *n* was exactly integral; this is because
+    the domain of the `nbinom` PMF includes only integral *n*, and the `nbinom`
+    object "knows" that. `nbinom` also knows that the shape *p* must be a
+    value between 0 and 1. In such a case - when the domain of the distribution
+    with respect to a parameter is finite - we are not required to specify
+    bounds for the parameter.
+
+    >>> shape_bounds = {'n': (0, 30)}  # omit parameter p using a `dict`
+    >>> res2 = stats.fit(dist, data, shape_bounds)
+    >>> np.testing.assert_allclose(res.params, res2.params)
+    True
+
+    If we wish to force the distribution to be fit with *n* fixed at 6, we can
+    set both the lower and upper bounds on *n* to 6. Note, however, that the
+    value of the objective function being optimized is worse (higher) in this
+    case.
+
+    >>> shape_bounds = {'n': (6, 6)}  # fix parameter `n`
+    >>> res3 = stats.fit(dist, data, shape_bounds)
+    >>> res3.params
+    FitShapes(n=6.0, p=0.5486556076755706, loc=0.0)  # may vary
+    >>> res3.objective_val > res.objective_val
+    True
+
+    Note that any `optimizer` parameter allows us to control the optimizer
+    used to perform the fitting. The default optimizer is
+    `scipy.optimize.differential_evolution` with its own default settings,
+    but we can easily change these settings by creating our own optimizer
+    with different defaults.
+
+    >>> from scipy.optimize import differential_evolution
+    >>> def optimizer(fun, bounds, *, integrality):
+    ...     return differential_evolution(fun, bounds, strategy='best2bin',
+    ...                                   seed=rng, integrality=integrality)
+    >>> shape_bounds = [(0, 30), (0, 1)]
+    >>> res4 = stats.fit(dist, data, shape_bounds, optimizer=optimizer)
+    >>> res4.params
+    FitShapes(n=5.0, p=0.5032774713044523, loc=0.0)  # may vary
+
+    """
+
+    # --- Input Validation / Standardization --- #
```

Comment:
Usually I put my input validation in a separate, private function. Here, I didn't think that was important, because almost all of this function _is_ input validation.

Reply:
I agree here.